python/nflxprofile/flamegraph.py

```diff
@@ -193,10 +194,25 @@ def process(self, stack):
         You probably want to avoid overriding this method. Override other
         methods to customize behavior instead.
         """
-        for frame in stack:
+        middle_out_filter = True
+        for i, frame in enumerate(stack):
             frame, frame_extras = self.process_frame(frame)
             if self.should_skip_frame(frame, frame_extras):
                 continue
+
+            if self.middle_out and middle_out_filter:
+                # middle out merge and no stack match yet
+                next_frame = stack[i + 1]
```

Comment:
What if `stack[i + 1]` is at top? Do we need a guard?

Reply:
True, I'll fix